### DISCUSSION TITLE
Conditional request execution support

### DISCUSSION THREAD

**Discussion by shadow1ng** [projectdiscovery/nuclei#2016] [Ideas]
发送poc时，能否按顺序发包和匹配，而不是一开始就发送完所有包。
比如下面两个poc
https://github.com/projectdiscovery/nuclei-templates/blob/master/vulnerabilities/ruijie/ruijie-eg-rce.yaml
https://github.com/chaitin/xray/blob/master/pocs/ruijie-eg-cli-rce.yml  

` .\xray_windows_amd64.exe webscan --plugins phantasm --poc "ruijie-eg-cli-rce.yml" -u 127.0.0.1`

使用xray时，会按poc规则顺序发包，如果第一个包匹配失败，就不会继续无效发包
![image](https://user-images.githubusercontent.com/39651966/163920856-e7a2c7c0-9c37-476c-b0f7-88fbb93b2c17.png)

` nuclei -u http://127.0.0.1 -t vulnerabilities\other\ruijie-eg-rce.yaml -proxy http://127.0.0.1:8080`
![image](https://user-images.githubusercontent.com/39651966/163920429-c3663392-23c3-445f-8060-f4aa135db318.png)
即使第一个数据包会匹配失败，但nuclei会一次性把所有包发完，再对他们进行匹配，导致nuclei浪费了很多发包，而且特征也更明显。

**Comment by ehsandeep** [comment]
**Translated version of the issue:**

When sending poc, can the packets be sent and matched in order instead of sending all the packets at the beginning.
For example, the following two pocs
https://github.com/projectdiscovery/nuclei-templates/blob/master/vulnerabilities/ruijie/ruijie-eg-rce.yaml
https://github.com/chaitin/xray/blob/master/pocs/ruijie-eg-cli-rce.yml

`.\xray_windows_amd64.exe webscan --plugins phantasm --poc "ruijie-eg-cli-rce.yml" -u 127.0.0.1`

When using xray, the packets will be sent in the order of the poc rules. If the first packet fails to match, it will not continue to send invalid packets.
![image](https://user-images.githubusercontent.com/39651966/163920856-e7a2c7c0-9c37-476c-b0f7-88fbb93b2c17.png)


`nuclei -u http://127.0.0.1 -t vulnerabilities\other\ruijie-eg-rce.yaml -proxy http://127.0.0.1:8080`

![image](https://user-images.githubusercontent.com/39651966/163920429-c3663392-23c3-445f-8060-f4aa135db318.png)

Even if the first packet fails to match, nuclei will send all the packets at one time, and then match them, resulting in nuclei wasting a lot of packets, and the characteristics are more obvious.

It is recommended that nuclei adopts the poc rule process of xray to send packets in order, thereby reducing invalid packets.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
